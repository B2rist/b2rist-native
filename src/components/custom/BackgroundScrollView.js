import React, {useContext} from 'react';
import {SafeAreaView, ScrollView, StatusBar} from 'react-native';
import {SettingsContext} from 'SettingsProvider';
import {Colors} from 'react-native/Libraries/NewAppScreen';

const BackgroundScrollView = ({children, dialog}) => {
  const {settings} = useContext(SettingsContext);
  const isDarkMode = settings.theme === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
    flex: 1,
  };

  return (
    <SafeAreaView style={backgroundStyle}>
      <StatusBar
        barStyle={isDarkMode ? 'light-content' : 'dark-content'}
        backgroundColor={backgroundStyle.backgroundColor}
      />
      <ScrollView
        contentInsetAdjustmentBehavior="automatic"
        style={backgroundStyle}>
        {children}
      </ScrollView>
      {dialog}
    </SafeAreaView>
  );
};

export default BackgroundScrollView;
