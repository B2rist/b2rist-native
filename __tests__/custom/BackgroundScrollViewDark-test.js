import 'react-native';
import React from 'react';
import {render, screen} from '@testing-library/react-native';
import BackgroundScrollView from 'components/custom/BackgroundScrollView';
import {View} from 'react-native';

jest.mock('react', () => {
  const ActualReact = jest.requireActual('react');
  return {
    ...ActualReact,
    useContext: () => ({
      settings: {
        theme: 'dark',
      },
    }),
  };
});

it('background scroll view dark mode test', () => {
  render(
    <BackgroundScrollView>
      <View />
    </BackgroundScrollView>,
  );
  expect(screen.toJSON()).toMatchSnapshot();
});
