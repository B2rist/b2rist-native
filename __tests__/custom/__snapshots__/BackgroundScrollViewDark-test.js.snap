// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`background scroll view dark mode test 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#222",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    style={
      {
        "backgroundColor": "#222",
        "flex": 1,
      }
    }
  >
    <View>
      <View />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
